cognite/client/data_classes/assets.py

```diff
@@ -204,21 +210,28 @@ def files(self) -> "FileMetadataList":
         return self._retrieve_related_resources(FileMetadataList, self._cognite_client.files)
 
     def _retrieve_related_resources(self, resource_list_class, resource_api):
+        seen = set()
+
+        def retrieve_and_deduplicate(asset_ids):
+            res = resource_api.list(asset_ids=asset_ids, limit=-1)
+            resources = resource_list_class([])
+            for resource in res:
+                with self._lock:
```

Comment:
Actually, move this lock outside the for loop. No point in releasing/acquiring the lock in every iteration 😅

Reply:
Yes, I thought so too.